mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessage.java

```diff
@@ -46,295 +48,415 @@
-        this.cc = EMailers.from(headerCollection.getCcAddressSet());
-        this.bcc = EMailers.from(headerCollection.getBccAddressSet());
-        this.sentDate = DateResolutionFormater.DATE_TIME_FOMATTER.format(headerCollection.getSentDate().orElse(internalDate));
-    }
-
-    private void copyMessageFields(MailboxMessage message, ZoneId zoneId) {
-        this.uid = message.getUid().asLong();
-        this.mailboxId = message.getMailboxId().serialize();
-        this.modSeq = message.getModSeq();
-        this.size = message.getFullContentOctets();
-        this.date = DateResolutionFormater.DATE_TIME_FOMATTER.format(getSanitizedInternalDate(message, zoneId));
-        this.mediaType = message.getMediaType();
-        this.subType = message.getSubType();
-        this.isAnswered = message.isAnswered();
-        this.isDeleted = message.isDeleted();
-        this.isDraft = message.isDraft();
-        this.isFlagged = message.isFlagged();
-        this.isRecent = message.isRecent();
-        this.isUnRead = ! message.isSeen();
-        this.userFlags = message.createFlags().getUserFlags();
-        this.properties = message.getProperties();
-    }
-
-    private static ZonedDateTime getSanitizedInternalDate(MailboxMessage message, ZoneId zoneId) {
-        if (message.getInternalDate() == null) {
-            return ZonedDateTime.now();
-        }
-        return ZonedDateTime.ofInstant(
-            Instant.ofEpochMilli(message.getInternalDate().getTime()),
-            zoneId);
-    }
-
-    private void generateText() {
-        this.text = Stream.of(from.serialize(),
-                to.serialize(),
-                cc.serialize(),
-                bcc.serialize(),
-                subjects.serialize(),
-                bodyText.orElse(null),
-                bodyHtml.orElse(null))
-            .filter(str -> !Strings.isNullOrEmpty(str))
-            .collect(Collectors.joining(" "));
-    }
-
-    private long uid;
-    private String mailboxId;
-    private List<String> users;
-    private long modSeq;
-    private long size;
-    private String date;
-    private String mediaType;
-    private String subType;
-    private boolean isUnRead;
-    private boolean isRecent;
-    private boolean isFlagged;
-    private boolean isDeleted;
-    private boolean isDraft;
-    private boolean isAnswered;
-    private String[] userFlags;
-    private Multimap<String, String> headers;
-    private EMailers from;
-    private EMailers to;
-    private EMailers cc;
-    private EMailers bcc;
-    private EMailers replyTo;
-    private Subjects subjects;
-    private String sentDate;
-    private List<Property> properties;
+    public static final SimpleProperty HAS_ATTACHMENT_PROPERTY = new SimpleProperty(PropertyBuilder.JAMES_INTERNALS, PropertyBuilder.HAS_ATTACHMENT, "true");
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
     private List<MimePart> attachments;
```

Comment:
Sorry to not have spotted it before.
You still have non final attributes. Why ? Could you make them final ?